templates/registration/two_factor_auth.html

```diff
@@ -0,0 +1,98 @@
+
+            function buf2hex(buffer) {
+                return Array.prototype.map.call(new Uint8Array(buffer), x => ('00' + x.toString(16)).slice(-2)).join('');
+            }
+
+            $('#use-webauthn').click(event => {
+                event.preventDefault();
+
+                if (typeof window.PublicKeyCredential === 'undefined') {
+                    alert('{{ _('WebAuthn is not supported by your browser.') }}');
+                    return;
+                }
+
+                $.getJSON('{{ url('webauthn_assert') }}')
+                    .done(publicKey => {
+                        decodeJSONBytes(publicKey);
+                        navigator.credentials.get({publicKey})
+                            .then(credential => {
+                                credential = {
+                                    id: credential.id,
+                                    response: {
+                                        authData: urlSafeBase64Encode(credential.response.authenticatorData),
+                                        clientData: urlSafeBase64Encode(credential.response.clientDataJSON),
+                                        signature: buf2hex(credential.response.signature),
+                                    }
+                                };
+                                $('#id_webauthn_response').val(JSON.stringify(credential));
+                                $('#2fa-form').submit();
+                            });
+                    })
+                    .fail(() => alert('{{ _('Failed to contact server.') }}'));
+            })
+        });
+    </script>
+{% endblock %}
+
+{% block body %}
+    <div id="login-panel-2fa">
+        <form action="" method="post" class="form-area" id="2fa-form">
+            {% csrf_token %}
+            {% if form.errors %}
+                <div id="form-errors">
+                    <p class="error">
+                        {%- if form.errors['totp_token'] -%}
+                            {{ ' '.join(form.errors['totp_token']) }}
+                        {%- else -%}
+                            {{ ' '.join(form.non_field_errors()) }}
+                        {%- endif -%}
+                    </p>
+                </div>
+            {% endif %}
+
+            {% if request.profile.is_totp_enabled %}
+                <div><label class="inline-header grayed">{{ _('Enter the 6-digit code generated by your app:') }}</label></div>
+                <div id="totp-token-container"><span class="fullwidth">{{ form.totp_token }}</span></div>
+                <hr>
+            {% endif %}
+            {% if request.profile.is_webauthn_enabled %}
+                {{ form.webauthn_response }}
+                <button style="display: inline" id="use-webauthn">{{ _('Use security key') }}</button>
+            {% endif %}
+            {% if request.profile.is_totp_enabled %}
+                <button style="float:right;" type="submit">{{ _('Login!') }}</button>
```

Comment:
```suggestion
                <button style="float:right;" type="submit">{{ _('Log in!') }}</button>
```

Reply:
This is also old code from `templates/registration/totp_auth.html`.

Reply:
You're still touching it so it's already going into git blame, so this is a good time for incremental improvements.